### DISCUSSION TITLE
PV outputs

### DISCUSSION THREAD

**Discussion by vmotakat** [GRIDAPPSD/gridappsd-forum#114] [API] [answered]
Hi
I checked inverter rating and actual power from glm file but my final platform results indicated that there was issue on inverter sizing. To validate my model I need to see all pv outputs from platform. To this end, I am looking for query to get PV outputs from platform. 

Do they have any such documentation let me know.
@poorva1209 @AAndersn

**Comment by AAndersn** [comment, marked as answer]
Please see https://gridappsd-training.readthedocs.io/en/latest/api_usage/3.3-Using-the-PowerGrid-Models-API.html#Querying-for-Measurements

Set the "objectType" key to "PowerElectronicsConnection"

```
from gridappsd import topics as t
topic = t.REQUEST_POWERGRID_DATA

message = {
    "requestType": "QUERY_OBJECT_MEASUREMENTS",
    "modelId": model_mrid,
    "objectType": "PowerElectronicsConnection",
    "resultFormat": "JSON"
}

gapps.get_response(topic, message)
```

**Reply by AAndersn** [reply]
You can then save the measurement ids and pass them to the simulation subscriber class. The PV output values will be included in the simulation message obtained by the on_message(headers, message) method

https://gridappsd-training.readthedocs.io/en/latest/api_usage/3.6-Controlling-Simulation-API.html#App-Core-as-a-Class-Definition
